crypto/bio/bss_file.c

```diff
@@ -349,8 +361,14 @@ static int file_gets(BIO *bp, char *buf, int size)
         if (!fgets(buf, size, (FILE *)bp->ptr))
             goto err;
     }
-    if (buf[0] != '\0')
-        ret = strlen(buf);
+    after = file_pos(bp);
+    if (after < before)
+        goto err;
+    if (after - before < size - 1)
```

Comment:
Are you sure these 2 lines are correct? (ftell is the location in the file)
If size = 3, before = 0
then after = 3

0123

Reply:
Yes, I am sure that the calculation is correct.
It is not hat easy to see, but the key is that fgets reads at most `size - 1` chars - see its man page:
```
fgets() reads in at most one less than size characters from stream and stores them into the  buffer  pointed  to  by  s.
```
Thus if `size` is 3 and `before` is 0 then `after` cannot be 3 but 2,
so `after - before` equals 2 equals `size - 1`.

Reply:
Maybe make this obvious, so that someone reading this code later doesn't try changing it..
/* a comment similar to above */
len = size - 1;

Reply:
I wont hold it up on this.. Will let you decide..

Reply:
Good point to add a comment. I've added:
```
/* fgets should guarantee that input length == after - before <= size - 1 */
```

Reply:
On this occasion I've squashed all fixup commits.
Thank you for approving, and so I ask you to re-approve regarding the new comment.

Reply:
Still approved.